pkg/storer/internal/reserve/reserve.go

```diff
@@ -197,6 +213,15 @@ func (r *Reserve) Put(ctx context.Context, chunk swarm.Chunk) error {
 
 		return nil
 	})
+	if err != nil {
+		if oldChunk != nil {
```

Comment:
And if the process terminates after the delete transaction and before this recovery, the original chunk is completely gone, right?

Reply:
fixed